raster-foundry/raster-foundry
# Update auth queries and endpoints to use text array ACRs

## Overview

This PR 
 - adds `authQuery` and `authorized` functions to the `ObjectPermissions` trait that do similar things as they were in the `Dao.scala` except that they use object level `acrs` now
 - changes related endpoints to use the above mentioned functions to determined if a user is authorized to list first class objects and to access anything, respectively
 - substitutes function calls from `AccessControlRuleDao` with those inherited from `ObjectPermissions` where it makes sense
 - adds property tests for the new sql command in `authQuery` and `authorized` functions
 - modifies existing property tests to use functions in `ObjectPermissions` instead

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~

### Notes

Before, when deactivating teams, we set `is_active = false` for any affected ACRs as well. This is not the case anymore. See issue #4020 for more details.

## Testing Instructions

 * Run migrations
 * Use the app frontend as you used to and do a bunch of different things - you should not feel like anything has changed really :joy:, but this is a good thing actually
 * CI
 * See in the code if there is anything I forgot to change

Closes #3922 
Closes #3923
